### DISCUSSION TITLE
Does fig-env option is supported with mermaid diagrams?

### DISCUSSION THREAD

**Discussion by jesustorresdev** [quarto-dev/quarto-cli#4832] [Q&A] [answered]
After read the documentation, I supposed that every fig- option work with diagrams.
But, in my case, fig-env is ignored with mermaid, even though it works with images.

I don't know if that case isn't supported or if I'm doing something wrong.

**Comment by cscheid** [comment, marked as answer]
We can't diagnose an issue without a minimal, full, example. (And the answer to your question depends on details which change depending on what your .qmd looks like)

**Reply by jesustorresdev** [reply]
Sorry, I only want to know if it should work before open an issue.
This example doesn't work:

~~~
---
format:
    pdf:
        keep-tex: true
        include-in-header: 
            text: |
                \usepackage{rotating}
---

```{mermaid}
%%| fig-env: sidewaysfigure
sequenceDiagram
  participant Alice
  participant Bob
  Alice->>John: Hello John, how are you?
  loop Healthcheck
    John->>John: Fight against hypochondria
  end
  Note right of John: Rational thoughts <br/>prevail!
  John-->>Alice: Great!
  John->>Bob: How about you?
  Bob-->>John: Jolly good!
```
~~~

The environment is figure in the generated tex but it should be sidewaysfigure

**Reply by cscheid** [reply]
No worries. But the thing is we won't know if it's an issue before seeing the code :)

**Reply by cscheid** [reply]
(This _does_ look like a bug!)

**Reply by cscheid** [reply]
There are _two_ bugs, in fact. Hold on, we're making a fix on `main`.

**Reply by cscheid** [reply]
Pull request is up, https://github.com/quarto-dev/quarto-cli/pull/4833

The fix should be on the prereleases tomorrow.

Thanks for the report, this needed fixing!

**Reply by jesustorresdev** [reply]
Thanks for such a quick response
